### PR TITLE
Use Microsoft.Data.SqlClient 2.1.0

### LINKED ISSUE #29: Separate build against Microsoft.Data.SqlClient?
Recently I tried to run `Microsoft.SqlServer.Types` against new SqlClient and found out that it is not possible due to fact that Microsoft re-declared attributes like `SqlUserDefinedTypeAttribute` in new provider. See more details [here](https://github.com/linq2db/linq2db/pull/1929#issuecomment-541396922).

Don't know if anybody will want to use `Microsoft.SqlServer.Types` with new provider (we are not end-users), so I propose to use this issue to see if it is a wanted feature.

### LINKED ISSUE #51: When using Microsoft.Data.SqlClient an InvalidCastException will be thrown
When using Microsoft.Data.SqlClient 2.0.1 the following exception will be generated

`[System.InvalidCastException] Unable to cast object of type 'Microsoft.SqlServer.Types.SqlHierarchyId' to type 'Microsoft.Data.SqlClient.Server.IBinarySerialize'.
`
once proper assembly redirection is in place

```
 if (assemblyName.FullName == "Microsoft.SqlServer.Types, Version=10.0.0.0, Culture=neutral, PublicKeyToken=89845dcd8080cc91")
            {
                var a = typeof(SqlGeography).Assembly;
                _logger.Info("Rebinding to " + a.FullName);
                return a;
            }

```

### PR DESCRIPTION
**Pull request by ErikEJ**
fixes #51
fixes #29

Design decisions:
- Name of package ?
- Use of strong naming (unclear why this was removed from the original package) ?

### CODE REVIEW

**Comment by dotMorten** [comment]
> Use of signing (unclear why this was removed from the original package) ?

@ErikEJ what do you mean?

File: src/Microsoft.SqlServer.Types.Tests/Geography/WktTests.cs [hunk 1]
```diff
@@ -1,5 +1,10 @@
 ﻿using Microsoft.SqlServer.Types.Tests.Geometry;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
+#if NETCOREAPP3_1
+using Microsoft.Data.SqlClient;
+#else
+using System.Data.SqlClient;
+#endif
```
Comment (dotMorten) [inline src/Microsoft.SqlServer.Types.Tests/Geography/WktTests.cs:7]: I don't understand this condition? Why would it be based on the target framework?

Comment (ErikEJ) [reply src/Microsoft.SqlServer.Types.Tests/Geography/WktTests.cs:7]: Just to be able to run tests with M.D.S. - no doubt you can think of a smarter way 😁

**Comment by ErikEJ** [comment]
Signing: you had signing enabled at some point, but then removed it soon after... Unsure why.

File: src/Microsoft.SqlServer.Types.M.D.S/Microsoft.SqlServer.Types.M.D.S.csproj [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    <Compile Include="..\Microsoft.SqlServer.Types\SqlGeometry.cs" Link="SqlGeometry.cs" />
+    <Compile Include="..\Microsoft.SqlServer.Types\SqlGeometryBuilder.cs" Link="SqlGeometryBuilder.cs" />
+    <Compile Include="..\Microsoft.SqlServer.Types\SqlHierarchyId.cs" Link="SqlHierarchyId.cs" />
+    <Compile Include="..\Microsoft.SqlServer.Types\SqlHierarchy\BitPattern.cs" Link="SqlHierarchy\BitPattern.cs" />
+    <Compile Include="..\Microsoft.SqlServer.Types\SqlHierarchy\BitReader.cs" Link="SqlHierarchy\BitReader.cs" />
+    <Compile Include="..\Microsoft.SqlServer.Types\SqlHierarchy\BitWriter.cs" Link="SqlHierarchy\BitWriter.cs" />
+    <Compile Include="..\Microsoft.SqlServer.Types\SqlHierarchy\HierarhyId.cs" Link="SqlHierarchy\HierarhyId.cs" />
+    <Compile Include="..\Microsoft.SqlServer.Types\SqlHierarchy\KnownPatterns.cs" Link="SqlHierarchy\KnownPatterns.cs" />
+    <Compile Include="..\Microsoft.SqlServer.Types\Wkt\WktReader.cs" Link="Wkt\WktReader.cs" />
+    <Compile Include="..\Microsoft.SqlServer.Types\Wkt\WktWriter.cs" Link="Wkt\WktWriter.cs" />
```
Comment (dotMorten) [inline src/Microsoft.SqlServer.Types.M.D.S/Microsoft.SqlServer.Types.M.D.S.csproj:50]: Better to use star include for all these entries:
` <Compile Include="..\Microsoft.SqlServer.Types\**\*.cs" Link="%(RecursiveDir)%(Filename)%(Extension)" />`

**Comment by dotMorten** [comment]
> Signing: you had signing enabled at some point, but then removed it soon after... Unsure why.

I digitally sign the nuget packages locally at a later stage before publishing

**Comment by ErikEJ** [comment]
Sorry, I mean strong naming!!

**Comment by dotMorten** [comment]
Aaah good call. Didn't even notice that went away

**Comment by ErikEJ** [comment]
I think it may be worth keeping it to improve the user experience (as far as I can tell from discussions in various GitHub issues)

**Comment by ErikEJ** [comment]
Should I add it back to the original project?

**Comment by ErikEJ** [comment]
Left strong signing enabled in the new project, should I also add it back to the original, or remove it altogether?

**Comment by ErikEJ** [comment]
Let me know if more changes are needed.
:smile:

**Comment by ErikEJ** [comment]
@dotMorten Anything I can do to help get this in?

**Pull request closed without merging**